chroma-manager/chroma_core/lib/service_config.py

```diff
@@ -342,29 +334,17 @@ def _stop_services(self):
                 log.error(error)
                 raise RuntimeError(error)
 
-        # Wait for supervisord to stop running
-        SUPERVISOR_STOP_TIMEOUT = 20
+        # Wait for 'iml-manager.target' to stop running
+        IML_MANAGER_STOP_TIMEOUT = 20
         t = 0
         stopped = False
         while True:
-            try:
-                SupervisorStatus().get_all_process_info()
-            except socket.error:
-                # No longer up
-                stopped = True
-            except xmlrpclib.Fault, e:
-                if (e.faultCode, e.faultString) == (6, 'SHUTDOWN_STATE'):
-                    # Up but shutting down
-                    pass
-                else:
-                    raise
-
-            if stopped:
+            if not ServiceControl.create('iml-manager.target').running:
                 break
             else:
-                if t > SUPERVISOR_STOP_TIMEOUT:
-                    raise RuntimeError("chroma-supervisor failed to stop after %s seconds" %
-                                       SUPERVISOR_STOP_TIMEOUT)
+                if t > IML_MANAGER_STOP_TIMEOUT:
```

Comment:
Will it succeed after > 1 tries if it fails the first time?

If so, that seems like something that needs to be fixed.  Don't want to mask issues here and now that we are putting all of this under the control of `systemd` it seems like a fine time to discover if there are issues which running > 1 times works around.